### PR TITLE
New record: Track_3_optimization: SOAP-MLP Contra/NorMuon 3148 steps

### PR DESCRIPTION
**Pull request by Srachuri-code**
## Summary

Adds a Track 3 optimization result at **3148 steps**, extending the SOAP/Shampoo-basis MLP preconditioning idea from PR #278 on top of the Contra-Muon + NorMuon-lite setup from PR #275.

This is 77 steps below the current merged Track 3 frontier at 3225 and 2 steps below the strongest live open Track 3 PR I found, PR #278 at 3150.

## Method

For MLP matrices (`mlp.fc.weight` and `mlp.proj.weight`), the Muon/Nesterov momentum update is first preconditioned in a SOAP/Shampoo eigenbasis, then passed through the same Contra-Muon + NorMuon-lite row/column variance normalization and u/w-floor postprocessing path.

Attention 2D matrices use the Contra-Muon + NorMuon-lite path without SOAP preconditioning. The archived scripts use the current upstream RMSNorm form and normal `logs/*.txt` capture files, with fixed per-run seeds and `train_steps = 3148`.

As in the Contra-Muon record, the scripts/logs print `muon_weight_decay=0.025`, but `Muon.step` does not use that param-group value, so effective Muon weight decay is zero.

## Reliability

16 non-cherry-picked seeds at K=3148:

| seed | val loss |
|---:|---:|
| 1 | 3.27888 |
| 2 | 3.27934 |
| 3 | 3.28172 |
| 4 | 3.27806 |
| 5 | 3.27840 |
| 6 | 3.27796 |
| 7 | 3.27715 |
| 8 | 3.27835 |
| 9 | 3.27760 |
| 10 | 3.27729 |
| 11 | 3.28080 |
| 12 | 3.27650 |
| 13 | 3.27822 |
| 14 | 3.27977 |
| 15 | 3.27832 |
| 16 | 3.27924 |
| **mean** | **3.27860000** |

Sample std: `0.00134744`

Track 3 significance rule:

```text
(3.28 - mean) * sqrt(n) >= 0.004
(3.28 - 3.27860000) * sqrt(16) = 0.00560000
```

## Rules

- Dataset, batch size, and architecture are unchanged.
- One forward/backward pass per step.
- No third-party optimizer library imports; optimizer code is inline in every logfile.
- No per-run val-loss early stopping; all submitted seeds use the same K=3148 stopping point.
- All launched 3148-step seeds for this protocol are included here.

## Files

- `records/track_3_optimization/README.md` adds the result-history row.
- `records/track_3_optimization/results/20260504_soap_mlp_contra_normuon_3148/README.md` summarizes the method and reliability calculation.
- `records/track_3_optimization/results/20260504_soap_mlp_contra_normuon_3148/seed01.txt` through `seed16.txt` are the submitted full-source logfiles.
- `records/track_3_optimization/results/20260504_soap_mlp_contra_normuon_3148/train_gpt_simple_soap_mlp_contra_normuon_3148.py` is a representative seed-1 train script extracted from `seed01.txt`.

## Validation

```bash
python3 -m optimizer_bench_autoresearch.cli audit --queue-id Q007
python3 -m optimizer_bench_autoresearch.cli score --group Q007 --write-json
python3 -m optimizer_bench_autoresearch.cli submission-packet --group Q007
python3 -m py_compile records/track_3_optimization/results/20260504_soap_mlp_contra_normuon_3148/train_gpt_simple_soap_mlp_contra_normuon_3148.py
git diff --cached --check
```

Local audit classification: `PASS_STATS_RULE_CLEAN`.

### CODE REVIEW

**Comment by KellerJordan** [comment]
it's nice to have extra seeds, but I'm a bit confused, what is the difference between this and #278?

**Comment by Srachuri-code** [comment]
sorry didn't mean to make this an official pr! just was curious about testing out what was done in 278 some more
